esphome/components/t6615/t6615.cpp

```diff
@@ -19,55 +19,66 @@ static const uint8_t T6615_COMMAND_ENABLE_ABC[] = {0xB7, 0x01};
 static const uint8_t T6615_COMMAND_DISABLE_ABC[] = {0xB7, 0x02};
 static const uint8_t T6615_COMMAND_SET_ELEVATION[] = {0x03, 0x0F};
 
+void T6615Component::send_ppm_command_() {
+  this->command_time_ = millis();
+  this->command_ = T6615Command::GET_PPM;
+  this->write_byte(T6615_MAGIC);
+  this->write_byte(T6615_ADDR_SENSOR);
+  this->write_byte(sizeof(T6615_COMMAND_GET_PPM));
+  this->write_array(T6615_COMMAND_GET_PPM, sizeof(T6615_COMMAND_GET_PPM));
+}
+
 void T6615Component::loop() {
-  if (!this->available())
+  if (this->available() < 5) {
+    if (this->command_ == T6615Command::GET_PPM && millis() - this->command_time_ > T6615_TIMEOUT) {
+      /* command got eaten, clear the buffer and fire another */
+      while (this->available())
+        this->read_byte(this->response_buffer_);
+      this->send_ppm_command_();
+    }
     return;
+  }
+
+  /* by the time we get here, we know we have at least five bytes in the buffer */
+  this->read_array(this->response_buffer_, 5);
```

Comment:
As `this->response_buffer_` is only used in this function, we can just put it on the stack instead:
```suggestion
  uint8_t response_buffer[6];
  this->read_array(this->response_buffer, 5);
```
(This also needs changes further down, and the removal of the `response_buffer_` class member).

Reply:
I don't think I need the `this->` in that case either, so I'll remove it.

Reply:
Yeah, definitely, you're right, I missed changing that in the suggestion.